lib/truffle/coverage.rb

```diff
@@ -10,7 +10,12 @@
 
 module Coverage
 
-  def self.start
+  def self.supported?(mode)
+    mode == :lines
+  end
+  
+  def self.start(*arguments, **options)
+    # Arguments/options are ignored.
```

Comment:
This doesn't seem correct, e.g. if `lines: false` we shouldn't enable coverage.

For unknown kwargs, should we raise ArgumentError?

Surprisingly CRuby seems to ignores extra kwargs:
```
$ ruby -v -rcoverage -e 'Coverage.start(foo: true)'   
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
```

But it does not allow any positional argument:
```
$ ruby -v -rcoverage -e 'Coverage.start(1)'        
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
-e:1:in `start': no implicit conversion of Integer into Hash (TypeError)

Coverage.start(1)
               ^
	from -e:1:in `<main>'

$ ruby -v -rcoverage -e 'Coverage.start(1, foo: true)'
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
-e:1:in `start': wrong number of arguments (given 2, expected 0..1) (ArgumentError)

Coverage.start(1, foo: true)
               ^^^^^^^^^^^^
	from -e:1:in `<main>'
```

Reply:
> This doesn't seem correct, e.g. if `lines: false` we shouldn't enable coverage.

Actually that seems the CRuby semantics :/
So that part seems OK.

```
$ ruby -v -rcoverage -e 'Coverage.start(foo: true); p Coverage.running?'
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]
true
```
(and also `ruby -v -rcoverage -e 'Coverage.start(foo: true); p Coverage.running?; require "drb"; p Coverage.result'`)

Reply:
Maybe we should write more ruby specs for this behaviour? I think we can tighten it up for 3.3.